cnd/migrations/2019-11-11-034058_create-message-tables/up.sql

```diff
@@ -0,0 +1,91 @@
+-- Your SQL goes here
+
+CREATE TABLE rfc003_bitcoin_ethereum_bitcoin_ether_request_messages
+(
+    id INTEGER               NOT NULL PRIMARY KEY,
+    swap_id UNIQUE           NOT NULL,
+    bitcoin_network          NOT NULL,
+    ethereum_chain_id        NOT NULL,
+    bitcoin_amount           NOT NULL,
+    ether_amount             NOT NULL,
+    hash_function            NOT NULL,
+    bitcoin_refund_identity  NOT NULL,
+    ethereum_redeem_identity NOT NULL,
+    bitcoin_expiry           NOT NULL,
+    ethereum_expiry          NOT NULL,
+    secret_hash              NOT NULL
+);
+
+CREATE TABLE rfc003_ethereum_bitcoin_ether_bitcoin_request_messages
+(
+    id INTEGER               NOT NULL PRIMARY KEY,
+    swap_id UNIQUE           NOT NULL,
+    bitcoin_network          NOT NULL,
+    ethereum_chain_id        NOT NULL,
+    bitcoin_amount           NOT NULL,
+    ether_amount             NOT NULL,
+    hash_function            NOT NULL,
+    bitcoin_redeem_identity  NOT NULL,
+    ethereum_refund_identity NOT NULL,
+    bitcoin_expiry           NOT NULL,
+    ethereum_expiry          NOT NULL,
+    secret_hash              NOT NULL
+);
+
+CREATE TABLE rfc003_bitcoin_ethereum_bitcoin_erc20_request_messages
+(
+    id INTEGER               NOT NULL PRIMARY KEY,
```

Comment:
Good idea. Since `swap_id` is `unique` it can also just serve as the `id`.

Reply:
This is actually very much on purpose. Having a primary key separate from any domain concepts makes it easier to establish foreign key relations.

Since that one is naturally ascending, it also helps the database in creating indicies and layout on disk.

I am very reluctant to changing this as I would like to keep the primary key an implementation detail of the database and not mix it with our data.